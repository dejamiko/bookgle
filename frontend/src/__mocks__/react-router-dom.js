module.exports = {
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn(),
  useGetUser: jest.fn(),
};