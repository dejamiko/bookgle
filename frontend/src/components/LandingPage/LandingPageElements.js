import styled from "styled-components";

export const WelcomeText = styled.text`
    font-family: 'Source Sans Pro', sans-serif;
    color: #000;
    opacity: 0.6;
    font-size: 20px;
    font-weight: 500;
    text-align: center;
`

export const LogoText = styled.text`
    font-family: 'Source Sans Pro', sans-serif;
    color: #000;
    font-size: 50px;
    font-weight: 500;
    line-height: 60px;
    margin-bottom: 5rem;
`

export const LoginText = styled.text`
    font-family: "Source Sans Pro";
    font-size: 15px;
    color: #0057FF;
`

export const ButtonContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 4rem;
`

export const RoundedButton = styled.button`
    border: 3px solid black;
    border-radius: 100px;
    height: 3rem;
    width: 12rem;
    background-color: #fff;
    margin-top: 1rem;
    font-family: 'Source Sans Pro', sans-serif;
    font-weight: 600;
    &:hover {
        border: 0px;
        background-color: #653FFD;
        color: #FFF;
    }
`

export const UnderLineText = styled.text`
    position: absolute;
    left: 12%;
    top: 16rem;
    width: 250px;
    height: 14px;
    transform: skew(-12deg) translateX(-50%);
    background: rgba(101,63,253,0.5);
    z-index: -1;
`

export const PurpleCircle = styled.div`
    height: 15rem;
    width: 15rem;
    position: absolute;
    border-radius: 100%;
    background-color: #653FFD;
    bottom: 38rem;
    left: 10rem;
    z-index: -2;
`

export const BlueCircle = styled.div`
    height: 11rem;
    width: 11rem;
    position: absolute;
    border-radius: 100%;
    background-color: #7BA8FF;
    bottom: 41rem;
    left: 3rem;
    z-index: -1;
`