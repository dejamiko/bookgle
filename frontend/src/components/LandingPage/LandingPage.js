import React from 'react'
import {Link} from 'react-router-dom'
import {Container, Row, Col, Button} from 'reactstrap'
import {BlueCircle, ButtonContainer, LoginText, LogoText, PurpleCircle, UnderLineText, WelcomeText} from './LandingPageElements'
import Nav from '../Nav/Nav'

const LandingPage = () => {
    return (
        <div 
            style={{ overflow: "hidden" }}>
            <Container fluid style={{ padding:"0px" }}>
                <Row style={{ marginBottom: "8rem" }}>
                    <Nav isAuthenticated={false}/>
                </Row>
                <Row>
                    <Col xs={2}>
                    </Col>
                    <Col xs={8}>
                        <LogoText>
                            Find people who<br />
                            <span style={{ fontWeight: 600 }}>vibe</span> with your<br />
                            bookshelf.
                            <UnderLineText />
                        </LogoText><br /><br />
                        <WelcomeText>
                            Welcome to Bookgle, a social media app<br />
                            where the bookworms of our society converge.
                        </WelcomeText>
                        <ButtonContainer>
                        <Link to="/sign_up/">
                            <Button
                                style={{
                                    backgroundColor: "#653FFD",
                                    borderRadius: "10px",
                                    padding: "0.75rem 2rem 0.75rem 2rem",
                                    fontFamily: "Source Sans Pro",
                                    color: "#fff",
                                    fontSize: "20px",
                                    marginBottom:"1rem"
                                }}>
                                Sign Up Today
                            </Button>
                        </Link>
                        <Link to="/log_in/">
                            <LoginText>Or log into an existing account.</LoginText>
                        </Link>
                        </ButtonContainer>
                    </Col>
                    <Col xs={2}>
                        <BlueCircle />
                        <PurpleCircle />
                    </Col>
                </Row>
            </Container>
        </div>
    )
}

export default LandingPage
