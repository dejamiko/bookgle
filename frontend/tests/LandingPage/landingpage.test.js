import React from "react"
import ReactDOM from "react-dom"
import LandingPage from "../../src/components/LandingPage/LandingPage"
import renderer from "react-test-renderer"
import { render, cleanup } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

afterEach(cleanup)

it("renders without crashing", () => {
    // const div = document.createElement("div")
    render(<LandingPage />, { wrapper: MemoryRouter })
})

it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><LandingPage /></MemoryRouter>).toJSON()
    expect(tree).toMatchSnapshot()
})