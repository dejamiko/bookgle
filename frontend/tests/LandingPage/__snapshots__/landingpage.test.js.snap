// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": "10rem",
        }
      }
    >
      <div
        className="col"
      />
      <div
        className="col"
      >
        <text
          className="sc-bdvvtL zooZt"
        >
          Welcome to
        </text>
        <br />
        <text
          className="sc-gsDKAQ kOppkk"
        >
          bookgle
        </text>
        <br />
        <a
          href="/log_in"
          onClick={[Function]}
        >
          <button
            className="sc-dkPtRN eZIjW"
          >
            LOG IN
          </button>
          <br />
        </a>
        <a
          href="/sign_up"
          onClick={[Function]}
        >
          <button
            className="sc-dkPtRN eZIjW"
          >
            SIGN UP
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
